site/src/pages/CreateWorkspacePage/CreateWorkspacePage.tsx

```diff
@@ -118,15 +117,15 @@ const CreateWorkspacePage: FC = () => {
       const newWorkspace = await autoCreateWorkspaceMutation.mutateAsync({
         templateName,
         organizationId,
-        defaultBuildParameters: autofillParameters,
-        defaultName: defaultName ?? generateWorkspaceName(),
+        buildParameters: autofillParameters,
+        name: defaultName ?? generateWorkspaceName(),
         versionId: realizedVersionId,
+        match: searchParams.get("match"),
       });
 
       onCreateWorkspace(newWorkspace);
     } catch (err) {
-      searchParams.delete("mode");
-      setSearchParams(searchParams);
+      setMode("form");
```

Comment:
I'm wondering if it might still be good to update the search params on top of the state change, but good catch with this

Reply:
I've thought about this and I don't think it makes much of a difference in the user experience, based on what I can see. Since no other part of the component uses it, I just removed it and decided to set the mode only in the state.